prometheus/process_collector_cgo_darwin.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build darwin && cgo
```

Comment:
Oh, you can mark things for CGO, nice!

Reply:
I had seen that before, but I didn't realize the build tag worked in the C files too.  Apparently it aggressively compiles any *.c files without waiting to see if cgo is used in *.go files, and complains if *.c files are present when `CGO_ENABLED=0`- unless the build tags are present in the *.c files.